packages/core-data/src/fetch/fetch-link-suggestions.ts

```diff
@@ -0,0 +1,356 @@
+	return results;
+}
+
+/**
+ * Sort search results by relevance to the given query.
+ *
+ * Sorting is necessary as we're querying multiple endpoints and merging the results. For example
+ * a taxonomy title might be more relevant than a post title, but by default taxonomy results will
+ * be ordered after all the (potentially irrelevant) post results.
+ *
+ * We use cosine similarity to determine relevance. This is a common technique used in information
+ * retrieval to determine how similar two documents are. In this case, we're treating the search
+ * query as a document and the title of the search result as another document.
+ *
+ * @param results
+ * @param search
+ */
+export function sortResults( results: SearchResult[], search: string ) {
+	const searchTerms = tokenize( search );
+	const searchTermFrequencies = getTermFrequencies( searchTerms );
+
+	const scores = {};
+	for ( const result of results ) {
+		const titleTerms = tokenize( result.title );
+		const titleTermFrequencies = getTermFrequencies( titleTerms );
+		scores[ result.id ] = getCosineSimilarity(
+			searchTermFrequencies,
+			titleTermFrequencies
+		);
+	}
+
+	return results.sort( ( a, b ) => scores[ b.id ] - scores[ a.id ] );
+}
+
+/**
+ * Turns text into an array of tokens.
+ *
+ * For example, `"I'm having a ball"` becomes `[ "im", "having", "a", "ball" ]`.
+ *
+ * @param text
+ */
+function tokenize( text: string ) {
+	return text
+		.toLowerCase()
+		.split( /\s+/ )
+		.map( ( word ) => word.replace( /\W/g, '' ) )
+		.filter( ( word ) => !! word );
+}
+
+/**
+ * Returns a map of term frequencies.
+ *
+ * For example "the cat in the hat" becomes `{ the: 2, cat: 1, in: 1, hat: 1 }`.
+ *
+ * The map can be interpreted as a vector in a multi-dimensional space, where
+ * each dimension corresponds to a unique word, and the value of the dimension
+ * is the frequency of that word.
+ *
+ * @param terms
+ */
+function getTermFrequencies( terms: string[] ) {
+	const frequencies = {};
+	for ( const term of terms ) {
+		frequencies[ term ] = ( frequencies[ term ] ?? 0 ) + 1;
+	}
+	return frequencies;
+}
+
+/**
+ * Returns the cosine similarity between two vectors.
+ *
+ * Cosine similarity measures the cosine of the angle between two vectors, which gives an
+ * indication of how similar the two vectors are.
+ *
+ * See https://en.wikipedia.org/wiki/Cosine_similarity for the mathematical definition.
+ *
+ * Cosine similarity ranges from -1 to 1:
+ *
+ * - A cosine similarity of 1 indicates that the two vectors are identical.
+ * - A cosine similarity of 0 indicates that the two vectors are orthogonal (no similarity).
+ * - A cosine similarity of -1 indicates that the two vectors are diametrically opposed.
+ *
+ * In the context of text similarity, a higher cosine similarity indicates that the documents share
+ * more common terms and are therefore more similar in content.
+ *
+ * @param vector1
+ * @param vector2
+ */
+function getCosineSimilarity(
```

Comment:
Is is worth adding unit tests for this in particular, or is it enough that it gets called as part of the test for `sortResults`?

Reply:
Yeah, why not. I can add simple unit tests for all the helper functions.

Reply:
I added more tests.